StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Link/LinkPaymentController.swift

```diff
@@ -0,0 +1,221 @@
+                        configuration: self.configuration,
+                        shouldOfferApplePay: false,
+                        shouldFinishOnClose: true,
+                        shouldPayInFlow: false
+                    )
+                    continuation.resume(returning: linkController)
+                case .failure(let error):
+                    continuation.resume(throwing: error)
+                }
+            }
+        }
+
+        linkController.payWithLinkDelegate = self
+        linkController.modalPresentationStyle = UIDevice.current.userInterfaceIdiom == .pad
+            ? .formSheet
+            : .overFullScreen
+
+        defer { linkController.dismiss(animated: true) }
+        try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<Void, Swift.Error>) in
+            payWithLinkContinuation = continuation
+            presentingViewController.present(linkController, animated: true)
+        }
+    }
+
+    /// Completes the Link payment or setup.
+    /// - Note: Once `completion` is called with a `.completed` result, this `LinkPaymentController` instance should no longer be used, as payment/setup intents should not be reused. Other results indicate cancellation or failure, and do not invalidate the instance.
+    /// - Parameter presentingViewController: The view controller used to present any view controllers required e.g. to authenticate the customer
+    /// - Parameter completion: Called with the result of the payment after any presented view controllers are dismissed
+    @_spi(LinkOnly) public func confirm(from presentingViewController: UIViewController, completion: @escaping (PaymentSheetResult) -> Void) -> Void {
+        Task {
+            do {
+                try await confirm(from: presentingViewController)
+                completion(.completed)
+            } catch Error.canceled {
+                completion(.canceled)
+            } catch {
+                completion(.failed(error: error))
+            }
+        }
+    }
+
+    /// Completes the Link payment or setup.
+    /// - Note: Once this method returns successfully, this `LinkPaymentController` instance should no longer be used, as payment/setup intents should not be reused. Thrown errors indicate cancellation or failure, and do not invalidate the instance.
+    /// - Parameter presentingViewController: The view controller used to present any view controllers required e.g. to authenticate the customer
+    /// - Throws: Either `LinkPaymentController.Error.canceled`, meaning the customer canceled the flow, or an error describing what went wrong.
+    @MainActor
+    @_spi(LinkOnly) public func confirm(from presentingViewController: UIViewController) async throws -> Void {
+        if (intent == nil || paymentOption == nil) && LinkAccountService().hasSessionCookie {
+            // If the customer has a Link cookie, `present` may not need to have been called - try to load here
```

Comment:
Interesting, this makes sense. How do merchants know if the customer has a link cookie?

Reply:
Great question, and one of the big unanswered ones around how merchants are actually going to _use_ this API. The user driving this product hasn't started work on their solution yet, so I'm in the dark as to what their expectations are currently. If they're tracking the customer's selection of Link on their end independently, they won't need to know anything; if not, though (or if there's some more complicated logic around it), there might be the need to add a `hasLinkAccount` property or similar to `LinkPaymentController` that they can check. I'm holding off on that until I have a better sense of what the merchant's needs are here.